feat: restricted Lorentz group equal to identity component

jnlindsay opened the pull request.

jnlindsay committed:
initial lemma statement

jnlindsay committed:
typo

jnlindsay committed:
orthochronous file: new lemmas; restricted file: finished second half of proof

jnlindsay committed:
simplified second half of `restricted_eq_identity_component`

jnlindsay commented on PhysLean/Relativity/Lorentz/Group/Orthochronous.lean:
```diff
@@ -219,6 +223,28 @@ lemma IsOrthochronous.iff_inv_isOrthochronous {Λ : LorentzGroup d} :
+  rw [IsOrthochronous.iff_in_orthchroRep_ker, IsOrthochronous.iff_in_orthchroRep_ker]
+  rw [MonoidHom.mem_ker, MonoidHom.mem_ker]
+  change orthchroMap Λ = 1 ↔ orthchroMap Λ' = 1
+  rw [h]
```
@jstoobysmith Any suggestions on what to name this lemma?

jstoobysmith replied:
Maybe we could call it something like `isOrthochronous_iff_of_orthchroMap_eq`

jnlindsay committed:
`·`

jnlindsay committed:
named lemma as `isOrthochronous_iff_of_orthchroMap_eq`

jnlindsay committed:
slicker proof; lots of `sorry`s

jnlindsay committed:
finished `restricted_eq_identity_component`

jnlindsay committed:
slight cleanup

jnlindsay committed:
reorganise

jnlindsay committed:
delete proof that won't work

jnlindsay committed:
now using SL(2, R) strategy

jstoobysmith committed:
Merge branch 'master' into jnlindsay/feat/restricted-Lorentz-group-equal-to-identity-component

jstoobysmith committed:
refactor: orthochronous lemmas and add semiformal results

copilot: Renamed several lemmas in Lorentz/Group/Orthochronous.lean for consistency with naming conventions. Added semiformal_result statements for key results in Boosts/Generalized.lean and Restricted.lean, including a formula for the time component of a generalized boost and the connectedness of the restricted Lorentz group. Removed an unused import in ToComplex.lean.

jstoobysmith committed:
refactor: Minkowski product and update references

Copilot: Replaces the old `innerProduct` definition for Lorentz vectors with a new `minkowskiProduct` and associated API, moving all references and lemmas to use the new name. Updates all dependent files to use `minkowskiProduct` and its lemmas, and removes the deprecated `innerProduct` code. Also renames and moves the Boosts module to a more appropriate location.

jstoobysmith committed:
refactor: Lint

jstoobysmith committed:
feat: Add Lorentz velocity

jstoobysmith committed:
refactor: Move Lorentz Group

jstoobysmith committed:
feat: Update semi-formal result

jstoobysmith committed:
refactor: Generalized boosts

jstoobysmith committed:
refactor: LInt
